### PR TITLE
Promises refactor 

### PR DESCRIPTION
**Pull request by ksgeorgieva**
- Add typings in 1.2 test files 
- Remove redundant async/awaits

### CODE REVIEW

**Review by robmoffat — DISMISSED** [review]

**Comment by robmoffat** [comment]
I've approved this, but I'd like to get @Joe-Dunleavy  and @kriswest 's opinions before we merge

**Review by robmoffat — COMMENTED** [review]
looks like there is some regression running these tests on Finsemble.  will take a look

File: src/test/v1.2/advanced/channels-support-1.2.ts
Comment (robmoffat) [inline src/test/v1.2/advanced/channels-support-1.2.ts:108]: to get this to work in Finsemble, I need to make this `async`

File: src/test/v1.2/advanced/channels-support-1.2.ts
Comment (robmoffat) [inline src/test/v1.2/advanced/channels-support-1.2.ts:113]: & here

File: src/test/v1.2/advanced/channels-support-1.2.ts
Comment (robmoffat) [inline src/test/v1.2/advanced/channels-support-1.2.ts:124]: same here

File: src/test/v1.2/advanced/channels-support-1.2.ts
Comment (robmoffat) [inline src/test/v1.2/advanced/channels-support-1.2.ts:129]: same here

**Comment by robmoffat** [comment]
@ksgeorgieva please review my comments.   I have no idea why these need to be async, but it seems like they do?

File: src/test/v1.2/advanced/channels-support-1.2.ts
Comment (kriswest) [reply src/test/v1.2/advanced/channels-support-1.2.ts:108]: Shouldn't need it to be... I don't have to use an async handler in either the prerelease version I'm running, nor 8.2 on either channel type... Both these are working for me:

```javascript
fdc3.addContextListener(null, (context) => {console.log("received context: ", context);})
```

```javascript
const channel = await fdc3.getOrCreateChannel("test");
channel.addContextListener(null, (context) => {console.log("received context via test: ", context);})
```

What output/error are you getting?

File: src/test/v1.2/advanced/fdc3.open.ts
Comment (kriswest) [inline src/test/v1.2/advanced/fdc3.open.ts:298]:
```suggestion
                ` Checking for current context of type "${contextType}" for test: "${testId}"
            Current context did ${context ? "" : "NOT "
```

File: src/test/v1.2/basic/fdc3.joinChannel.ts
Comment (kriswest) [inline src/test/v1.2/basic/fdc3.joinChannel.ts:50]: duplicated code is back again
```suggestion
```

File: directories/local-conformance-1_2.v2.json
Comment (Joe-Dunleavy) [inline directories/local-conformance-1_2.v2.json:26]: The logo should be sourced from https://www.scottlogic.com/sites/default/files/scott-logic-favicon.jpg.
Same for all of the other ones in this file

File: directories/local-conformance-2_0.v2.json
Comment (Joe-Dunleavy) [inline directories/local-conformance-2_0.v2.json:26]: same thing here for the logo source

**Review by kriswest — CHANGES_REQUESTED** [review]
Duplicated code has crept in again ((added a suggestion). Also commented on issue you (@robmoffat) had with sync context handler (need more info, should not need to be async)...

Otherwise this is looking good

File: src/test/v1.2/advanced/fdc3.open.ts
Comment (robmoffat) [reply src/test/v1.2/advanced/fdc3.open.ts:298]: ok, well I'm not the arbiter of truth on what is good for Finsemble - happy to merge if @kriswest is.

File: src/test/v1.2/basic/fdc3.joinChannel.ts
Comment (ksgeorgieva) [reply src/test/v1.2/basic/fdc3.joinChannel.ts:50]: This branch was created on top of Rob's PR. I'll update it since it's accepted and merged

File: src/test/v1.2/advanced/channels-support-1.2.ts
Comment (kriswest) [reply src/test/v1.2/advanced/channels-support-1.2.ts:108]: We went through this and determined that the test is actually broken - adding async was somehow shuffle the order of things so it appeared to work/pass but the test is no longer being implemented as defined due to the lack of a typed context listener here

**Pull request closed without merging**
